client/app/admin/admin.component.ts

```diff
@@ -0,0 +1,188 @@
+
+import { AdminService } from './admin.service';
+import { MenuService } from '../menu/menu.service';
+import { MenuItemCategory, MenuItem } from '../menu/menu.model';
+
+@Component({
+    selector: 'app-admin',
+    templateUrl: './admin.component.html',
+    styleUrls: ['./admin.component.css']
+})
+export class AdminComponent implements OnInit {
+    newMenuForm = this.formBuilder.group({
+        name: ['', Validators.required],
+        price: [0, Validators.required],
+        category: ['Main']
+    });
+
+    public menu = [];
+
+    public staff = [
+        {
+            firstname: 'Klaus',
+            surname: 'Logan',
+            role: 'Waiter'
+        },
+        {
+            firstname: 'Susanne',
+            surname: 'Smith',
+            role: 'Waiter'
+        },
+        {
+            firstname: 'Joanne',
+            surname: 'Lee',
+            role: 'Manager'
+        },
+        {
+            firstname: 'John',
+            surname: 'McGregor',
+            role: 'Manager'
+        },
+        {
+            firstname: 'Lucy',
+            surname: 'McCabe',
+            role: 'Kitchen'
+        },
+        {
+            firstname: 'James',
+            surname: 'Peter',
+            role: 'Kitchen'
+        },
+        {
+            firstname: 'Amanda',
+            surname: 'Hogen',
+            role: 'Waiter'
+        },
+        {
+            firstname: 'Lewis',
+            surname: 'Marino',
+            role: 'Waiter'
+        }
+    ];
+
+    private displayOrdersTodayBool;
+    private displayOrdersWeekBool;
+    private displayOrdersMonthBool;
+
+    constructor(
+        protected adminService: AdminService,
+        private menuService: MenuService,
+        private formBuilder: FormBuilder
+    ) {}
+
+    ngOnInit() {
+        this.getMenu();
+    }
+
+    getMenu() {
+        this.menuService.getMenu().subscribe(result => {
+            this.menu = result;
+        });
+    }
+
+    updateMenu() {
+        var table = document.getElementById("menu tr");
+        for (var i = 1, row; row = table[i]; i++) {
+            //iterate through rows
+            //rows would be accessed using the "row" variable assigned in the for loop
+            for (var j = 0, col; col = row[j]; j++) {
+                console.log(col[j]);
+            }
+        }
+        //this.menuService.update(itemId, menuItem);
+    }
+
+    addItem() {
+        if (this.newMenuForm.valid) {
+            var newItem = this.newMenuForm.value;
+            console.log(newItem);
+            this.menuService.create(newItem)
+                .subscribe((data) => {
+                    console.log('item created');
+                    console.log(data.item)
+                }, () => {
+                    console.log('failed');
+                });
+        }
+       
+    }
+
+    removeItem(id_) {
+        var element = document.getElementById(id_);
+        element.remove();
+        // this.menuItem.remove(id_);
```

Comment:
For this to communicate with the api, we will need to add a new method to `/client/app/menu/menu.service.ts`  similar to 
```
delete(id: string) {
        return this.httpClient.delete(
            environment.apiUrl + 'menu/' + id
        );
    }
```
and then within this method call the service method 
```
this.menuService.delete(id_).subscribe(() => {
    console.log('deleted');
})
```